src/uri/uriService.ts

```diff
@@ -0,0 +1,96 @@
+import type { Disposable, Event, Uri, UriHandler } from 'vscode';
+import { EventEmitter, window } from 'vscode';
+import type { Container } from '../container';
+import type { DeepLinkType } from '../deepLink/deepLink';
+import { DeepLinkTypes } from '../deepLink/deepLink';
+import { log } from '../system/decorators/log';
+import type { DeepLinkUriEvent, UriEvent } from './uri';
+import { UriTypes } from './uri';
+
+// This service is in charge of registering a URI handler and handling/emitting URI events received by GitLens.
+// URI events to GitLens take the form of: vscode://eamodio.gitlens/... and are handled by the UriEventHandler.
+// The UriEventHandler is responsible for parsing the URI and emitting the event to the UriService.
+export class UriService implements Disposable {
+	private _disposable: Disposable;
+	private _uriHandler: UriHandler = { handleUri: this.handleUri.bind(this) };
+	private _uriEventEmitter: EventEmitter<UriEvent> = new EventEmitter<UriEvent>();
+
+	constructor(private readonly container: Container) {
+		this._disposable = window.registerUriHandler(this._uriHandler);
+	}
+
+	dispose() {
+		this._disposable.dispose();
+	}
+
+	get onUri(): Event<UriEvent> {
+		return this._uriEventEmitter.event;
+	}
+
+	// Set up a deep link event based on the following specifications:
+	// 1. Remote link type: /repolink/{repoId}?url={remoteUrl}
+	// 2. Branch link type: /repolink/{repoId}/branch/{branchName}?url={remoteUrl}
+	// 3. Tag link type: /repolink/{repoId}/tag/{tagName}?url={remoteUrl}
+	// 4. Commit link type: /repolink/{repoId}/commit/{commitSha}?url={remoteUrl}
+	// If the url does not fit any of the above specifications, return null
+	// If the url does fit one of the above specifications, return the deep link event
+	private formatDeepLinkUriEvent(uri: Uri): DeepLinkUriEvent | null {
+		const uriSplit = uri.path.split('/');
```

Comment:
I can see `URLSearchParams` replacing the `parseQuery` part, but would you mind clarifying what `URL` offers to make parsing the path easier? i.e. the `/link/r/{repoId}/b/{branchName}` part.

Reply:
Sorry that was for the parseQuery (it was late) 🙃

Reply:
Updated in commit: [Use URLSearchParams instead of parseQuery](https://github.com/gitkraken/vscode-gitlens/pull/2471/commits/08660d91b2a2acbae68fa3d55567b57d9b1ea3f4)